Overview.src.html

```diff
@@ -392,11 +392,26 @@ <h2 class="no-num no-toc">
 &lt;script&gt;
   // the cast button is visible if at least one presentation display is available
   var castBtn = document.getElementById("castBtn");
-  // availablechange event is fired when the presentation display availability changes
-  navigator.presentation.onavailablechange = function (evt) {
-    // show or hide cast button depending on display availability
-    castBtn.style.display = evt.available ? "inline" : "none";
+  // the availability listener object.
+  var availability = null;
+  // show or hide cast button depending on display availability
+  var handleAvailabilityChange = function(available) {
+    castBtn.style.display = available ? "inline" : "none";
   };
+  // Promise is resolved as soon as the presentation display availability is known.
+  navigator.presentation.getAvailability().then(function(availabilityObj) {
+    // availability.current may be kept up-to-date by the UA as long as the availability
+    // object is alive. It is advised for the web developers to discard of the object
+    // as soon as it's not needed (see the starting a new presentation example below).
+    availability = availabilityObj;
```

Comment:
You don't need that. You could simply have ```availability``` as the argument in the then statement, like:
```js
navigator.presentation.getAvailability().then(function(availability) {
  // use |availability|.
});
```

Calling ```.onchange``` on the object should keep it alive as long as ```onchange``` isn't reset to ```null```.